src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -71,19 +89,20 @@ func (client *Client) ListenAndServe(onListeningFn func()) error {
 		ErrorLog:     log.AsStdLogger(),
 	}
 
-	log.Debugf("About to start client (HTTP) proxy at %s", client.Addr)
-
+	log.Debugf("About to start client (HTTP) proxy at %s", listenAddr)
 	return httpServer.Serve(l)
 }
 
 // Configure updates the client's configuration. Configure can be called
-// before or after ListenAndServe, and can be called multiple times.  It
-// returns the highest QOS fronted.Dialer available, or nil if none available.
-func (client *Client) Configure(cfg *ClientConfig) {
+// before or after ListenAndServe, and can be called multiple times.
+func (client *Client) Configure(cfg *ClientConfig, proxyAll func() bool) {
 	client.cfgMutex.Lock()
 	defer client.cfgMutex.Unlock()
 
 	log.Debug("Configure() called")
+	if client.bal == nil {
+		client.bal = eventual.NewValue()
```

Comment:
As noted in the commit, I wasn't able to do this.  In fact, I had to move the initialization of the variable even earlier since it comes into use as soon as a user agent connects to the client proxy, which can happen before Configure gets called.